app/grandchallenge/components/backends/docker.py

```diff
@@ -252,11 +254,20 @@ def _provision_input_volume(self):
             self._copy_input_files(writer=writer)
 
     def _copy_input_files(self, writer):
-        for file in self._input_files:
+        for input_file in self._input_files:
+            if isinstance(input_file, tuple):
+                name, input_file = input_file
+                if not hasattr(input_file, "name"):
+                    file = NamedTemporaryFile(delete=True)
+                    file.write(bytes(str(input_file), "utf-8"))
```

Comment:
This serializes all of the values to text. I don't think this is what we want, I think we want to serialize as JSON, so that it can be loaded in the algorithm with something like (in L14 `copyio.py`):

```
    files = {x for x in Path("/input").rglob("*") if x.is_file()}

    for file in files:
        try:
            with open(file) as f:
                val = json.loads(f.read())
        except Exception as e:
            warn(f"Could not load {file} as json, {e}")
            continue

        res[str(file.absolute())] = val
```

Otherwise, converting from stringified Python to types is going to be a minefield for the user.